Added some test for aggregation and optional match

SaschaPeukert opened the pull request:
The original problem seems to be fixed but there was no actual test.
Thus I added one for mixing optional and count(distinct n.prop)

SaschaPeukert committed:
Added some test for aggregation and optional match

The original problem seems to be fixed but there was no actual test.
Thus I added one for mixing optional and count(distinct n.prop)

alexaverbuch commented on enterprise/cypher/acceptance-spec-suite/src/test/resources/cypher/features/PatternPredicates.feature:
```diff
@@ -304,6 +304,23 @@ Feature: PatternPredicates
       | (:A)  |
     And no side effects
 
+  Scenario: Using a optional match after aggregation and before an aggregation
```
I think this scenario should be moved to : https://github.com/SaschaPeukert/neo4j/blob/6af43b6ff2a740b37f71b9c75a1318809f30a825/enterprise/cypher/acceptance-spec-suite/src/test/resources/cypher/features/OptionalMatchAcceptance.feature
(1) there is no pattern predicate in the scenario, e.g., `WHERE <pattern>`
(2) because that is an Optional Match-specific feature file

SaschaPeukert replied:
I created a new AggregationAcceptance.feature file and added the test to that.

SaschaPeukert committed:
Moved test to new AggregationAcceptance.feature

Also fixed the name of the SkipLimitAcceptance feature

SaschaPeukert committed:
Moved test to new AggregationAcceptance.feature

Also fixed the name of the SkipLimitAcceptance feature.
(This time including the actual changes...)
